### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by dimarogkov**
[DEMO LINK](https://dimarogkov.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,94 @@
...
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```
Comment (etojeDenys) [inline src/components/NewMovie/NewMovie.tsx:18]: we can combine it in one useState
```
const [newMovie, setNewMovie] = useState({
  title: '',
  ...
})
```

now we can also create only one handle function for all fields
```
const handleChange = (e: ...) => {
  const {name, value} = e.target;
  setMovie((prevMovie) => ({...prevMovie, [name]: value}))
}
```

don't forget to specify name attr to input in TextField

**Review by etojeDenys — CHANGES_REQUESTED** [review]
let's show the error when the user enters only spaces in field

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,95 @@
...
+    onAdd(newMovie);
+    resetForm();
+    setCount(prev => prev + 1);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
```
Comment (DarkMistyRoom) [inline src/components/NewMovie/NewMovie.tsx:51]: no need to set `key` here. Use it only for the lists

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
please show the errors when inputs are filled with white spaces only. If you need help with it, please ask in the chat

**Review by maxim2310 — CHANGES_REQUESTED** [review]
please show the errors when inputs are filled with white spaces only. If you need help with it, please ask in the chat

**Review by etojeDenys — CHANGES_REQUESTED** [review]
still not fixed
image

**Comment by dimarogkov** [comment]
Name at least one reason why I should take away what should be on condition?

**Comment by dimarogkov** [comment]
image

**Review by VitaliyBondarenko1982 — APPROVED** [review]
GJ!

**Pull request closed without merging**